Clean and rationalize Child theme page templates

matt-bernhardt opened the pull request:
This attempts to rationalize and clean the principle page templates in the child theme, including the content partials which they include. The first commit removes a few templates which are no longer needed, while the remaining commits each focus on one page template and associated partials. Individual commit messages try to summarize what is happening in each, but many of them do the same sort of things:

* Make sure the file comments at the top of each template are accurate
* Improve code readability via smaller PHP blocks, less indenting, and avoiding complex conditionals
* Shrink the size of The Loop wherever possible, so that things like repeated `id` attributes don't happen (and to improve readability)
* Make better use of HTML comments when long divs close - using constructs like `<!-- end div#NAME -->`

While pursuing these cleaning steps, it became clear that three of the page templates were very similar - with their relevant distinctions being whether they invoked various sidebars. In further discussions with UX, we decided that using page templates in this way introduced unneeded complexity, as the question of "where will this widget appear?" became spread across widget settings and the templates each page used.

We have thus marked two templates as DEPRECATED, with the hope of removing them entirely once all site content has been imported.

At this stage, the primary page templates provided by the Child theme are:
* Standard Child (page.php) - should be the default template for Page records.
* Front (page-front.php) - includes a loop of sticky Post records above the page record, useful for site-level announcements.
* Post Feed (page-posts-feed.php) - replaces the Below Content Widget area with a paginated display of Post records, useful for sites which need their own "recent news" page.

The deprecated templates are:
* Full Width (page-full-width.php) - identical to Standard Child, but does not load the right sidebar
* Widgetized (page-widgetized.php) - no difference from the Standard Child, which now supports the Below Content Widget Area

Beyond these primary templates, there are also some page templates that are specific to the Exhibits site. I've paid less attention to these templates in this work, other than to remove two retired templates that are no longer necessary.

## Developer

### Secrets

- [ ] All new secrets have been added to Pantheon tiers
- [ ] Relevant secrets have been updated in Github Actions
- [ ] All new secrets documented in README
- [x] No new secrets are created

### Documentation

- [ ] Project documentation has been updated
- [x] No documentation changes are needed

### Accessibility

- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)

### Stakeholder approval

- [x] Stakeholder approval has been confirmed
- [ ] Stakeholder approval is not needed

### Dependencies

NO dependencies are updated


## Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] The changes have been verified
- [ ] The documentation has been updated or is unnecessary
- [ ] New dependencies are appropriate or there were no changes

matt-bernhardt committed:
Clean the Widgetized template (page and content)

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  documented in what they offer, and their structure is not always
  rational (or comparable between themselves)

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

This reorganizes the Widgetized template (both the overall Page and the
content partial which it calls).

* Marks the Widgetized template as deprecated
* Conditional display of the breadcrumb area
* Clearer file block comments
* More tightly enclose calls to WP functions
* Precalculate the classes to be applied to div#content
* Reduced use of conditionals, and tightly enclose the relevant
  differences. Also, reduce the use of else blocks in conditionals
* Mark the end of the #stage and #content divs

** Document any side effects to this change:

* None, hopefully.

matt-bernhardt committed:
Rationalize the Posts Feed template

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  distinguished in what they offer, and at times have poorly constructed
  logic and code flow.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

This reorganizes the Posts Feed template, along with the associated
inc/content-feed.php partial. In general terms, it attempts to make
these templates similar to the other templates in the theme, to allow
their relevant differences to stand out.

Specific changes include:
* Improved file comments
* Pursuing the same cleaning used in other commits in this branch, which
  pushes some markup and logic inside The Loop from the page template
  to the content-feed partial.
* Notably, this page template does _NOT_ implement the 'Below Content
  Widget Area' sidebar, because that area below the page body is
  instead used (within content-feed) to query recent posts and iterate
  over that recordset.
* Shrink the size of The Loop to surround the content-feed partial -
  similar to other template cleaning. This loop should still only be
  called once, but keeping it small helps keep the page template clean.
* Shrink the PHP blocks that just call WP functions
* Removes blank lines, and reduce indenting

** Document any side effects to this change:

* Hopefully there should be none, but the ultimate proof of this will
  be an inspection of the one page record (in the AKDC site) which makes
  use of this template. If that page looks okay and passes UX review,
  then we will know that this works as intended. If not, we will open
  more tickets to fix whatever is wrong.
* The div#sticky-posts-2 block in inc/content-feed.php should be
  avoided in this review, lest you potentially awake a sleeping dragon.
  I've carefully avoided changing it to make clear that whatever it is
  doing before, it is still doing now.

matt-bernhardt committed:
Remove unused templates

** Why are these changes being introduced:

* There are some page templates in this theme which are no longer
  needed, now that there is a generic Exhibits Location template.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

* This removes four templates which are no longer needed: two location-
  specific page templates which were superceded by the general Exhibits
  Location template, and then two category-specific templates which were
  part of an even earlier attempt at having a template that lists
  exhibits for a given location.

** Document any side effects to this change:

* None

matt-bernhardt committed:
Rationalize Standard Child template

This also resolves some mistakes in nesting that would have occurred had this template ever somehow returned multiple records.

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  documented in what they offer, and their structure is not always
  rational (or comparable between themselves)

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

* Clearer code documentation, including the file block at the top
* Reduce the size of The Loop for clarity and to avoid repeating markup
* More tightly enclose calls to WP functions
* Precalculate content_classes for cleaner echoing
* Add support for sidebar-two (the Below Content Widget Area), to enable
  us to standardize on the use of this template while deprecating some
  others.
* Avoid overwriting a sensitive $title variable in the content partial
* Reduced indenting

** Document any side effects to this change:

* None, hopefully

matt-bernhardt committed:
Clean the default template (index.php)

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  distinguished in what they offer, and at times have poorly constructed
  logic and code flow.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

This reorganizes the Default template (index.php)

* Adds an explicit Template Name to make sure that the template is
  available for selection in the page edit interface.
* Tightly enclose calls to theme functions rather than have a long block
  of PHP code.

** Document any side effects to this change:

* None, hopefully.

matt-bernhardt committed:
Clean the Front Page template

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  distinguished in what they offer, and their structure is not always
  rational (or comparable between themselves)

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

This makes the same sort of changes to this template that can be found
in other commits, trying to standardize them as much as possible.

* Clearer file doc comment
* Precalculate $content_classes variable, rather than using a conditional
  inside an HTML attribute on lines 40-45
* Shrink the size of The Loop, so any iteration contains only needed
  markup (not repeating div#content, for example)
* More tightly enclose PHP blocks which call PHP functions (calls to
  get_header( 'child' ) near the top of the page, for example)
* Reduced and more rational indenting

** Document any side effects to this change:

* None, hopefully

matt-bernhardt committed:
Clean the Full Width page template

** Why are these changes being introduced:

* The various page templates in the Child theme are not always clearly
  documented in what they offer, and their structure is not always
  rational (or comparable between themselves)

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/lm-142

** How does this address that need:

* Mark the template as Deprecated (we hope to remove this in a future
  release)
* Conditional display of breadcrumb div (either front page or standard
  variants)
* Improve code documentation (in case we end up needing to keep it)
* More tightly enclose calls to WP functions, and a smaller Loop
* Mark the end of #content and #stage divs
* Decreased and more standardized indenting

** Document any side effects to this change:

* None, hopefully.

matt-bernhardt commented:
The four violations being raised by CodeClimate appear to be unavoidable - they are noting code repetition at the top of some page template files, focusing specifically on lines that basically _need_ to be there. I'm particularly confused, though, about why _only_ four violations are being flagged, since more than four of these files are being touched.

JPrevost commented on web/app/themes/mitlib-child/inc/content-widgetized.php:
```diff
@@ -13,17 +15,20 @@
+		 * The "sidebar-two" region is named "Below Content Widget Area" within
+		 *  the admin interface.
+		 */
 		dynamic_sidebar( 'sidebar-two' );
```
Is the only difference between this and `inc/content-page.php` that the `dynamic_sidebar` is not conditional here and tries to display even if not active?

matt-bernhardt replied:
That conditional doesn't actually seem needed - `dynamic_sidebar()` returns nothing if there are no active widgets in that sidebar, so the returned markup is identical in that regard.

The only difference between these two partials at the moment is that `inc/content-page.php` adds a `content-main` class to one of the divs, so you get `<div class="main-content content-main">` instead of `<div class="main-content">`. This does seem relevant to some CSS rules around the site, but I'm not sure if this is something that UX will want to preserve or not.

JPrevost replied:
Interesting. One CSS class difference definitely feels like a "this template is gonna get booted" kinda a think. Later is fine though :)

JPrevost commented on web/app/themes/mitlib-child/templates/page.php:
```diff
@@ -2,63 +2,54 @@
+ * from the Parent theme.
+ *
+ * Please note that while this template does load the Main Sidebar defined by
+ * the Child theme, it does _not_ load the "Below Content Widget Area" sidebar.
```
Is the `Below Content Widget Area` not conditional-able like the other sidebar? I'm curious if staff can activate something in that area but not have them show up because the template doesn't support it similar to the confusion trying to be eliminated in some of the other work in this PR. If content in that section is different, that's fine but at this point I don't fully understand why we are doing a dedicated template for one type of sidebar and having the other be conditional based on what the content team has activated. (Not asking for a change so much as understanding)

matt-bernhardt replied:
That widget area is conditional-able, yes. It looks like I need to update this file comment, because the `content-page.php` partial it calls on line 43 actually _does_ load that area now. I forgot to update this - will push a new commit.

matt-bernhardt committed:
Update file comment for Standard Child template

** Why are these changes being introduced:

* The structure of the Standard Child page template changed after I
  edited the file comment for this template, and now needs to be updated

** How does this address that need:

* This updates the file comment for the Standard Page template, making
  it clear that this is now the preferred page template for the theme,
  and that it supports both widget areas defined in the theme.

** Document any side effects to this change:

* None
